clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryExecutor.java

```diff
@@ -0,0 +1,103 @@
+package com.linkedin.venice.datarecovery;
+
+import java.io.Console;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * DataRecoveryExecutor is the engine to run tasks in data recovery.
+ */
+public class DataRecoveryExecutor {
+  private final Logger LOGGER = LogManager.getLogger(DataRecoveryExecutor.class);
+  private final static int DEFAULT_POOL_SIZE = 10;
+  private final static int DEFAULT_POOL_TIMEOUT_IN_SECONDS = 30;
+  private final int poolSize;
+  private final ExecutorService pool;
+  private List<DataRecoveryTask> tasks;
+
+  public DataRecoveryExecutor() {
+    this(DEFAULT_POOL_SIZE);
+  }
+
+  public DataRecoveryExecutor(int poolSize) {
+    this.poolSize = poolSize;
+    this.pool = Executors.newFixedThreadPool(this.poolSize);
+  }
+
+  public void perform(Set<String> storeNames, StoreRepushCommand.Params params) {
+    String pass = getUserCredentials();
+    if (pass == null) {
+      LOGGER.error("Cannot get password, exit");
+      return;
+    }
+    params.setPassword(pass);
+
+    tasks = buildTasks(storeNames, params);
+    List<CompletableFuture<Void>> taskFutures = tasks.stream()
+        .map(dataRecoveryTask -> CompletableFuture.runAsync(dataRecoveryTask, pool))
+        .collect(Collectors.toList());
+    taskFutures.stream().map(CompletableFuture::join).collect(Collectors.toList());
+    displayTaskResult();
+
+    shutdownAndAwaitTermination();
+  }
+
+  private void shutdownAndAwaitTermination() {
+    pool.shutdownNow();
+    try {
+      if (!pool.awaitTermination(DEFAULT_POOL_TIMEOUT_IN_SECONDS, TimeUnit.SECONDS)) {
+        // Cancel currently executing tasks.
+        pool.shutdownNow();
```

Comment:
Shall we just throw exception here since `shutdownNow` function call would return immediately?

Reply:
If it throws, I am not sure how its caller should handle it?  
Can I follow some existing pattern in our code base? e.g. 

```
IsolatedIngestionServer.stopInner()
{...
    longRunningTaskExecutor.shutdown();
    try {
      if (!longRunningTaskExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
        longRunningTaskExecutor.shutdownNow();
      }
    } catch (InterruptedException e) {
      currentThread().interrupt();
    } 
...
}
```

Reply:
Sure, so the intent is trying to wait for the graceful shutdown and if it times out, interrupt the running task by `shutdownNow`.
Let us follow the existing pattern then.

Reply:
That's right, that's sounds like the intention. Cool, will fix it.